parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -919,6 +895,59 @@ public void appendRowGroup(SeekableInputStream from, BlockMetaData rowGroup,
     endBlock();
   }
 
+  /**
+   * Merges adjacent row groups in the supplied files while maintaining that the new groups is no more than the specified
+   * maxRowGroupSize
+   * @param inputFiles input files to merge
+   * @param maxRowGroupSize the maximum size in bytes the new created groups can be
+   * @param useV2Writer whether to use a V2 encoding based writer when rewriting dictionary encoded pages
+   * @param compression compression to use when writing
+   * @throws IOException
+   */
+  public void mergeRowGroups(List<InputFile> inputFiles, long maxRowGroupSize, boolean useV2Writer, CompressionCodecName compression) throws IOException {
+    new RowGroupMerger(schema, compression, useV2Writer).merge(inputFiles, maxRowGroupSize, this);
+  }
+
+  private List<ColumnChunkMetaData> getColumnsInOrder(BlockMetaData rowGroup, boolean dropColumns) {
```

Comment:
Converting to Entry and then back to List would add some cost, comparing with the original implementation. Can you keep it as List<ColumnChunkMetaData> in line of 922 for original use and wrap it up with List<Entry<ColumnDescriptor, ColumnChunkMetaData>> for new methods to use?